### PR TITLE
Remove deprecated Init(), pass instance.Settings around.

### LINKED ISSUE #10720: libbeat cmds do not respect `instance.Settings`
Several commands in `libbeat/cmd` do not respect `instance.Settings`, but only use dedicated attributes from it. This can lead to bugs for e.g. running `export template` etc. 
The bug was originally found in https://github.com/elastic/apm-server/issues/1922.

Ensure that the `settings` are passed on to all cmds and remove deprecated [`Init()`](https://github.com/elastic/beats/blob/master/libbeat/cmd/instance/beat.go#L240) .

### PR DESCRIPTION
**Pull request by simitt**
Ensure settings from rootCmd are respected also when calling `export` and `setup`.
fixes #10720

### CODE REVIEW

**Comment by simitt** [comment]
Failing tests are unrelated afaik.

**Comment by kvch** [comment]
`libbeat/cmd/version.go` does not use `instance.Settings` right now. Is that intentional?

File: libbeat/cmd/instance/beat.go [hunk 1]
```diff
@@ -681,6 +681,7 @@ func (b *Beat) loadDashboards(ctx context.Context, force bool) error {
 	}
 
 	if b.Config.Dashboards.Enabled() {
+
```
Comment (kvch) [inline libbeat/cmd/instance/beat.go]: Please remove this line.

**Comment by simitt** [comment]
While the `versionCmd` and the `completionCmd` do not need the `settings` info for their use case, I think it would be cleaner to also pass in the `settings` object to be aligned. Will change it, thanks for pointing out @kvch .

**Comment by kvch** [comment]
Thank you!

**Comment by simitt** [comment]
jenkins, retest this please

**Pull request merged**